components/entity-access-rules/api/src/main/java/org/phenotips/data/permissions/script/PermissionsManagerScriptService.java

```diff
@@ -112,12 +114,17 @@ public AccessLevel resolveAccessLevel(String name)
      *
      * @param targetPatient the patient
      */
-    public PatientAccess getPatientAccess(Patient targetPatient)
+    public EntityAccess getPatientAccess(Patient targetPatient)
     {
         return this.getPatientAccess(targetPatient.getId());
     }
 
-    public PatientAccess getPatientAccess(String targetPatientId)
+    public EntityAccess getEntityAccess(@Nullable final PrimaryEntity targetEntity)
+    {
+        return targetEntity != null ? getEntityAccess(targetEntity.getId()) : null;
```

Comment:
This seems very inefficient, to convert an entity to an ID, and then back to entity in `getEntityAccess(String id)`. I think this should be implemented the other way around, i.e. `getEntityAccess(String id)` should resolve the ID to an object, and then cal _this_ method instead to perform actual work.